### DISCUSSION TITLE
CGAL 5.0.4. reference to 'list' is ambiguous error

### DISCUSSION THREAD

**Discussion by ivabacic** [CGAL/cgal#6325] [Q&A] [answered]
I'm using gcc 9.3.0, boost 1.64.0 and CGAL 5.0.4. on Scientific Linux 6.9.

Getting many errors like this:
`In file included from /home/software/gcc/9.3.0/include/c++/9.3.0/list:63,
                 from /home/iva/CGAL-5.0.4/include/CGAL/Intersections_2/Iso_rectangle_2_Triangle_2.h:23,
                 from /home/iva/CGAL-5.0.4/include/CGAL/intersection_2.h:37,
                 from /home/iva/CGAL-5.0.4/include/CGAL/Kernel/function_objects.h:28,
                 from /home/iva/CGAL-5.0.4/include/CGAL/Cartesian/function_objects.h:20,
                 from /home/iva/CGAL-5.0.4/include/CGAL/Cartesian/Cartesian_base.h:62,
                 from /home/iva/CGAL-5.0.4/include/CGAL/Simple_cartesian.h:20,
                 from LinearPhysNets.cpp:10:
/home/software/gcc/9.3.0/include/c++/9.3.0/bits/stl_list.h:1991:22: error: reference to 'list' is ambiguous
 1991 |     operator==(const list<_Tp, _Alloc>& __x, const list<_Tp, _Alloc>& __y)
      |                      ^~~~
/home/software/gcc/9.3.0/include/c++/9.3.0/bits/stl_list.h:552:11: note: candidates are: 'template<class _Tp, class _Alloc> class std::__cxx11::list'
  552 |     class list : protected _List_base<_Tp, _Alloc>
      |           ^~~~`

Edited to add [full error list](https://gist.github.com/ivabacic/f8fa0e8025d63f826882b3f1a40a3ae5)

After many many hours of searching for the answer, I'm at loss how to proceed. What causes these errors?

**Comment by sloriot** [comment]
Are you stuck with CGAL 5.0.4? Could you try CGAL 5.4?

Seems to be an issue with boost and your stl. Try defining BOOST_DETAIL_NO_CONTAINER_FWD before including any boost/CGAL header

**Reply by ivabacic** [reply]
Unfortunately, I am stuck with that version of CGAL (I use a cluster which I cannot fully configure myself). I added the line 
`#define BOOST_DETAIL_NO_CONTAINER_FWD
`
to the top of my cpp file but I keep getting the same errors.

**Comment by sloriot** [comment, marked as answer]
I guess it is not defined early enough, cause you can't get an error about something ambiguous that is defined in a portion of the boost header guarded by this macro. Have a  look at `/usr/include/boost/detail/container_fwd.hpp`, maybe the macro is not named as in my header.

**Reply by ivabacic** [reply]
Problem solved. An old version of boost was being linked while compiling, all good now, thank you for pointing me in the right direction.
